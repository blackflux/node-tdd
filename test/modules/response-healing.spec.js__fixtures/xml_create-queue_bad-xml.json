{
  "fn": "call",
  "params": [
    "sqs:CreateQueueCommand",
    {
      "QueueName": "some-queue-name",
      "Attributes": {
        "DelaySeconds": "0",
        "MaximumMessageSize": "262144",
        "MessageRetentionPeriod": "345600",
        "ReceiveMessageWaitTimeSeconds": "0",
        "VisibilityTimeout": "360",
        "RedrivePolicy": "{\"deadLetterTargetArn\":\"arn:aws:sqs:us-west-2:012345678910:some-queue-arn\",\"maxReceiveCount\":10}"
      },
      "tags": {
        "tagA": "33a8baaf-6aed-4abc-95eb-95e40a8f4fa9",
        "tagB": "92769396-e725-452e-b56c-969b765bb731",
        "lastModified": "2022-06-27T21:08:54.033Z"
      }
    }
  ],
  "cassette": [
    {
      "scope": "https://sqs.us-west-2.amazonaws.com:443",
      "method": "POST",
      "path": "/",
      "body": "QueueName=some-queue-name&Tag.1.Key=tagA&Tag.1.Value=33a8baaf-6aed-4abc-95eb-95e40a8f4fa9&Tag.2.Key=tagB&Tag.2.Value=92769396-e725-452e-b56c-969b765bb731&Tag.3.Key=lastModified&Tag.3.Value=2022-06-27T21%3A08%3A54.033Z&Attribute.1.Name=DelaySeconds&Attribute.1.Value=0&Attribute.2.Name=MaximumMessageSize&Attribute.2.Value=262144&Attribute.3.Name=MessageRetentionPeriod&Attribute.3.Value=345600&Attribute.4.Name=ReceiveMessageWaitTimeSeconds&Attribute.4.Value=0&Attribute.5.Name=VisibilityTimeout&Attribute.5.Value=360&Attribute.6.Name=RedrivePolicy&Attribute.6.Value=%7B%22deadLetterTargetArn%22%3A%22arn%3Aaws%3Asqs%3Aus-west-2%3A799715345196%3Asome-queue-arn%22%2C%22maxReceiveCount%22%3A10%7D&Action=CreateQueue&Version=2012-11-05",
      "status": 200,
      "response": "<?xml some bad xml",
      "responseIsBinary": false
    }
  ]
}
