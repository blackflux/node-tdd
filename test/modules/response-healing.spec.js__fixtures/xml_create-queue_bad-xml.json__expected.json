[
  {
    "scope": "https://sqs.us-west-2.amazonaws.com:443",
    "method": "POST",
    "path": "/",
    "body": {
      "Attributes": {
        "DelaySeconds": "0",
        "MaximumMessageSize": "262144",
        "MessageRetentionPeriod": "345600",
        "ReceiveMessageWaitTimeSeconds": "0",
        "VisibilityTimeout": "360",
        "RedrivePolicy": "{\"deadLetterTargetArn\":\"arn:aws:sqs:us-west-2:012345678910:some-queue-arn\",\"maxReceiveCount\":10}"
      },
      "QueueName": "some-queue-name",
      "tags": {
        "tagA": "33a8baaf-6aed-4abc-95eb-95e40a8f4fa9",
        "tagB": "92769396-e725-452e-b56c-969b765bb731",
        "lastModified": "2022-06-27T21:08:54.033Z"
      }
    },
    "status": 200,
    "response": "<?xml some bad xml",
    "responseIsBinary": false,
    "reqheaders": {
      "amz-sdk-invocation-id": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$",
      "amz-sdk-request": "attempt=1; max=3",
      "authorization": "^AWS4-HMAC-SHA256 Credential=.+$",
      "content-length": "^[1-9][0-9]*$",
      "content-type": "application/x-amz-json-1.0",
      "host": "sqs.us-west-2.amazonaws.com",
      "user-agent": "^aws-sdk-js/.+$",
      "x-amz-content-sha256": "^[a-f0-9]{64}$",
      "x-amz-date": "^[0-9]{8}T[0-9]{6}Z$",
      "x-amz-target": "AmazonSQS.CreateQueue",
      "x-amz-user-agent": "^aws-sdk-js/[0-9]+.[0-9]+.[0-9]+$",
      "x-amzn-query-mode": "true"
    }
  }
]
